Yafc.Model/Model/Project.cs

```diff
@@ -71,16 +77,41 @@ protected internal override void ThisChanged(bool visualOnly) {
         metaInfoChanged?.Invoke();
     }
 
-    public static Project ReadFromFile(string path, ErrorCollector collector) {
+    public static Project ReadFromFile(string path, ErrorCollector collector, bool useMostRecent) {
         Project? project;
 
+        // Check whether there is an autosave that is saved at a later time than the current save.
+        if (useMostRecent) {
+            var savetime = File.GetLastWriteTimeUtc(path);
+            var highestAutosave = Enumerable
+                .Range(1, AutosaveRollingLimit)
+                .Select(i => new {
+                        Path = GenerateAutosavePath(path, i),
+                        Index = i,
+                        LastWriteTimeUtc = File.GetLastWriteTimeUtc(GenerateAutosavePath(path, i))
+                    }
+                )
+                .MaxBy(s => s.LastWriteTimeUtc);
+
+            if (highestAutosave != null && highestAutosave.LastWriteTimeUtc > savetime) {
+                path = highestAutosave.Path;
+            }
```

Comment:
Should we also update `autosaveIndex` so YAFC continue with the correct autosave number after loading? (You are already storing `Index`, so I guess we can just use this?)